openfl/component/collaborator/collaborator.py

```diff
@@ -409,17 +409,22 @@ def get_data_for_tensorkey(self, tensor_key):
                         creates_model=True,
                     )
                     self.tensor_db.cache_tensor({new_model_tk: nparray})
-                else:
-                    logger.info(
-                        "Count not find previous model layer."
-                        "Fetching latest layer from aggregator"
-                    )
-                    # The original model tensor should be fetched from client
-                    nparray = self.get_aggregated_tensor_from_aggregator(
-                        tensor_key, require_lossless=True
-                    )
```

Comment:
I don't understand this change - why does the `else` statement go outside?

Reply:
I'm not able to reproduce it, might be a transient failure I had. I'll revert this change and handle collaborator restarts in different pr if needed.